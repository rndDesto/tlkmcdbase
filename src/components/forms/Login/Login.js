import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import TextField from '../../fields/Text';
import Password from '../../fields/Password';
import styles from './styles.scoped.css';
import { useSelector } from 'react-redux';

export default function Login(props) {
  const { handleSubmit, pristine, submitting } = props;
  const { message } = useSelector(s => s.login);
  const { isLoading } = useSelector(s => s.loading);
  const textInputProps = { placeholder: 'cth. admin@email.com' };
  const passwordInputProps = { placeholder: 'Min. 6 karakter' };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={TextField} inputProps={textInputProps} label="Email" name="email"/>
      <Field component={Password} inputProps={passwordInputProps} label="Kata Sandi" name="password" />
      {message && <p className={styles.error}>{message}</p>}
      <Button disabled={pristine || submitting} onSubmit={handleSubmit}>
        { isLoading ? 'Menunggu...' : 'MASUK' }
      </Button>
    </form>
  );
}

Login.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired
};
