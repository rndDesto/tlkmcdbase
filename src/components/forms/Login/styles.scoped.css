.root > button {
  margin-top: 4rem;
  width: 100%;
}

.error {
  margin-top: 0.5rem;
  color: var(--palette-textfield-error-main);
}