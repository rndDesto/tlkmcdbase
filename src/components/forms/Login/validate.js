export default function validate(values) {
  const { email, password } = values;

  return {
    email: !email && 'required',
    password: !password && 'required',
  };
}
