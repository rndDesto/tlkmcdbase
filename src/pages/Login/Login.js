import React, { useState } from 'react';
import DataButton from '../../components/elements/DataButton';

const Login = () => {
  const [isLoading, setIsLoading]= useState(false);
  const mantul = ()=>{
    setIsLoading(true);
    setTimeout(()=>{
      setIsLoading(false);
    },5000);
  };
  return (
    <div>
      Login
      <img src="/assets/ic-close.svg" />
      <DataButton
        buttonProps={{ color:'primary', variant:'contained', disabled:isLoading }}
        id="mantul"
        isLoading={isLoading}
        onClick={mantul}>ini button</DataButton>
    </div>
  );
};

export default Login;
