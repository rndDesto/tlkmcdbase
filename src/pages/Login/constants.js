export const FAILED = 'Login/failed';
