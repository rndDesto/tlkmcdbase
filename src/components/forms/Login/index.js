import { reduxForm } from 'redux-form';
import Component from './Login';
import validate from './validate';

export default reduxForm({
  form: 'login',
  initialValues: {
    email:'',
    password:''
  },
  validate,
})(Component);
