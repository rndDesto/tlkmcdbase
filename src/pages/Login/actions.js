import { setToken, setUserData, setExpireTime } from '../../utils/storage';
import { loginUser } from '../../utils/fetch';
import { FAILED, LOADING } from './constants';

export function login(data) {
  return dispatch => {
    dispatch(loadingAction(true));

    loginUser(data)
      .then(res => {
        dispatch(loadingAction(false));
        if (res.data.accessToken) {
          const { accessToken } = res.data;
          setToken(accessToken);
          setUserData(res.data);
          setExpireTime(res.data.exp);
          location.href = '/';
        } else {
          dispatch(loginFailedAction('You are not allowed to access'));
        }
      })
      .catch(err => {
        const message = err.statusCode >= 400 && err.statusCode < 500 ? err.message : 'Username atau password yang ada masukan salah';
        dispatch(loginFailedAction(message));
        dispatch(loadingAction(false));
      });
  };
}

function loadingAction(isLoading) {
  return { type: LOADING, isLoading };
}

function loginFailedAction(message) {
  return { type: FAILED, message };
}
