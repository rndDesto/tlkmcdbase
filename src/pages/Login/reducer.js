import { FAILED, LOADING } from './constants';

const initialState = {
  isLoading: false,
};

export default function reducer(state = initialState, action) {
  const { type, isLoading, message } = action;

  switch (type) {
    case FAILED:
      return {
        ...state,
        message,
      };

    case LOADING:
      return {
        ...state,
        isLoading,
      };
    default:
      return state;
  }
}
