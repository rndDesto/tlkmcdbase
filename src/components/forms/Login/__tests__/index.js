import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Login from '../Login';
import { useSelector } from 'react-redux';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    login: {
      email: '',
      password: ''
    },
    message: 'test'
  };
});

describe('src/components/forms/Login', () => {
  let props = {
    handleSubmit: jest.fn(),
    pristine: true,
    submitting: false
  };

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Login {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
