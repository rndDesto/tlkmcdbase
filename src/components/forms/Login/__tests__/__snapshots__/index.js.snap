// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/Login render 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <ForwardRef
    component={[Function]}
    inputProps={
      Object {
        "placeholder": "cth. admin@email.com",
      }
    }
    label="Email"
    name="email"
  />
  <ForwardRef
    component={[Function]}
    inputProps={
      Object {
        "placeholder": "Min. 6 karakter",
      }
    }
    label="Kata Sandi"
    name="password"
  />
  <p>
    test
  </p>
  <Button
    buttonProps={Object {}}
    className=""
    disabled={true}
    onClick={[Function]}
    onSubmit={[MockFunction]}
  >
    MASUK
  </Button>
</form>
`;
